dpdispatcher/machine.py

```diff
@@ -286,9 +294,16 @@ def gen_script_command(self, job):
     def gen_script_end(self, job):
         job_tag_finished = job.job_hash + '_job_tag_finished'
         flag_if_job_task_fail = job.job_hash + '_flag_if_job_task_fail'
+
+        append_script_part = ""
+        append_script = job.resources.append_script
+        for ii in append_script:
+            append_script_part += "\n{ii}"
```

Comment:
SO SORRY for this severe typo and it should have been fixed ASAP!

Reply:
Well, it might be my ignorance that not using `string.join()` for I just thought error was raised when empty list passed. It also hint me that the unittests not include a tempelate rendering test, it might be a good idea to add one.